src/server/models/listItem/listItem.ts

```diff
@@ -0,0 +1,343 @@
+import { get, merge } from "lodash";
+import pgescape from "pg-escape";
+import { prisma } from "./../db/prisma-client";
+import { logger } from "server/services/logger";
+import {
+  LawyersFormWebhookData,
+  CovidTestSupplierFormWebhookData,
+  WebhookData,
+} from "server/components/formRunner";
+import {
+  List,
+  User,
+  ListItem,
+  ServiceType,
+  ListItemGetObject,
+  Point,
+} from "./../types";
+import { recordListItemEvent } from "./../audit";
+import { CovidTestSupplierListItem, LawyerListItem } from "./providers";
+import { ListItemWithAddressCountry } from "./providers/types";
+import { Prisma } from "@prisma/client";
+import { makeAddressGeoLocationString } from "server/models/listItem/geoHelpers";
+import { getChangedAddressFields } from "./providers/helpers";
+import { geoLocatePlaceByText } from "server/services/location";
+import { rawUpdateGeoLocation } from "server/models/helpers";
+
+export async function findListItemsForList(list: List): Promise<ListItem[]> {
```

Comment:
consider having a generic find or fetch function that accepts filter criteria and has conditionals around the values we permit filtering on, e.g., list item id.  The specific use-case function can then simply set this criteria and pass into the generic function.  This will then avoid having to duplicate the SQL in multiple places, e.g., currently defined in `providers/helpers.js fetchPublishedListItemQuery() `and `listItem/listItem.ts fetchPublishedListItemQuery()`, maybe other places